azurelinuxagent/common/protocol/wire.py

```diff
@@ -911,6 +917,43 @@ def get_header_for_cert(self):
             "x-ms-guest-agent-public-x509-cert": cert
         }
 
+    def get_host_plugin(self):
+        if self.host_plugin is None:
+            self.host_plugin = HostPluginProtocol(self.endpoint, self.get_goal_state())
+        return self.host_plugin
```

Comment:
Yes, we need to support changes in goal state (I guess change in HostPlugin endpoint does not really make much sense) and it is done in Update_goal_state()

Every preset interval, ExtHandlersHandler.run(),
```js
     def run(self):
         self.ext_handlers, etag = None, None
         try:
             self.protocol = self.protocol_util.get_protocol()
             self.ext_handlers, etag = self.protocol.get_ext_handlers()
         except ProtocolError as e:
```
gets executed which triggers protocol.get_ext_handlers(), 
```js
    def get_ext_handlers(self):
         logger.verbose("Get extension handler config")
         # Update goal state to get latest extensions config
         self.client.update_goal_state()
         goal_state = self.client.get_goal_state()
         ext_conf = self.client.get_ext_conf()
         # In wire protocol, incarnation is equivalent to ETag
         return ext_conf.ext_handlers, goal_state.incarnation
```
then WireClient.update_goal_state(). 

```js
@@ -702,6 +706,8 @@ def update_goal_state(self, forced=False, max_retry=3):
                  self.update_shared_conf(goal_state)
                  self.update_certs(goal_state)
                  self.update_ext_conf(goal_state)
 +                if self.host_plugin is not None:
 +                    self.host_plugin.goal_state = goal_state
                  return
```

This will effectively update goal_state ref of host_plugin.